platform/pulp/plugin/changeset/model.py

```diff
@@ -0,0 +1,182 @@
+    """
+
+    def __init__(self, model):
+        """
+        Args:
+            model (Model): A remote (wanted) model object.
+        """
+        self.model = model
+
+    def save(self):
+        """
+        Save the model.
+        """
+        self.model.save()
+        log.debug(_('Model: %(m)s - saved'), {'m': self})
+
+
+class RemoteContent(Remote):
+    """
+    Represents content contained within the remote repository but is not
+    contained within the local (pulp) repository.
+
+    Attributes:
+        model (pulp.plugin.Content): A real content model object.
+        settled (bool): Indicates that all artifacts have been downloaded
+            and the content is ready to be imported.
+
+        _artifacts (set): The set of related `RemoteArtifact`.
+    """
+
+    def __init__(self, model):
+        """
+        Args:
+            model (pulp.plugin.Content): A real content model object.
+
+        """
+        super(RemoteContent, self).__init__(model)
+        self._artifacts = set()
+        self.settled = False
+
+    @property
+    def key(self):
+        """
+        The natural key as a dictionary.
+
+        Returns:
+            dict: The natural key.
+        """
+        return {f.name: getattr(self.model, f.name) for f in self.model.natural_key_fields}
+
+    @property
+    def artifacts(self):
+        """
+        Associated artifacts (read only).
+
+        Returns:
+            frozenset: Associated artifacts.
+        """
+        return frozenset(self._artifacts)
+
+    def add_artifact(self, artifact):
+        """
+        Add a remote artifact.
+
+        Args:
+            artifact(RemoteArtifact): An artifact to add.
+        """
+        artifact.content = self
+        self._artifacts.add(artifact)
+
+    def update(self, model):
+        """
+        Update this `model` stored with the specified model that has been
+        fetched from the database. The artifacts are matched by `relative_path`
+        and their model object is replaced by the fetched model.
+
+        Args:
+            model (pulp.plugin.Content): A fetched content model object.
+        """
+        self.model = model
+        known = {a.model.relative_path: a for a in self._artifacts}
+        self._artifacts.clear()
+        for artifact in model.artifacts.all():
+            try:
+                found = known[artifact.relative_path]
+                found.model = artifact
+                self._artifacts.add(found)
+            except KeyError:
+                log.error(_('Artifact not matched.'))
+
+    def settle(self):
+        """
+        Settle matters that are prerequisite to associating the content
+        with a repository.  This mainly pertains to the artifacts being downloaded;
+        and associated artifact models saved; the content model saved.
+
+        Notes:
+            Called whenever an artifact has been downloaded.
+        """
+        for artifact in self._artifacts:
```

Comment:
Just as a fun alternative, this could be

```
if all((artifact.model.downloaded for artifact in self._artifacts)):
  self.save()
  self.settled=True
```

It may not be any better, but I do like the way it conveys that the conditional is a requirement for all things to be true, and the last two lines only happen if that condition is met.